src/main/java/org/veupathdb/service/eda/ds/plugin/alphadiv/AlphaDivBoxplotPlugin.java

```diff
@@ -0,0 +1,147 @@
+      );
+    requestedStreamsList.add(
+      new StreamSpec(COMPUTE_STREAM_NAME, computeConfig.getCollectionVariable().getEntityId())
+        .addVars(getChildrenVariables(computeConfig.getCollectionVariable()) 
+      ));
+    
+    return requestedStreamsList;
+  }
+
+  @Override
+ protected void writeResults(OutputStream out, Map<String, InputStream> dataStreams) throws IOException {
+    AlphaDivComputeConfig computeConfig = getComputeConfig();
+    AlphaDivBoxplotSpec spec = getPluginSpec();
+    Map<String, VariableSpec> varMap = new HashMap<>();
+    varMap.put("xAxisVariable", spec.getXAxisVariable());
+    varMap.put("overlayVariable", spec.getOverlayVariable());
+    varMap.put("facetVariable1", getVariableSpecFromList(spec.getFacetVariable(), 0));
+    varMap.put("facetVariable2", getVariableSpecFromList(spec.getFacetVariable(), 1));
+    String showMissingness = spec.getShowMissingness() != null ? spec.getShowMissingness().getValue() : "FALSE";
+    String computeStats = spec.getComputeStats() != null ? spec.getComputeStats().getValue() : "FALSE";
+    String showMean = spec.getMean() != null ? spec.getMean().getValue() : "FALSE";
+    String valueSpec = spec.getValueSpec().getValue();
+    String computeEntityPKColName = toColNameOrEmpty(getComputeEntityPrimaryKeyVarSpec(computeConfig.getCollectionVariable().getEntityId()));
+    
+    useRConnectionWithRemoteFiles(dataStreams, connection -> {
+      List<VariableSpec> computeInputVars = ListBuilder.asList(getComputeEntityPrimaryKeyVarSpec(computeConfig.getCollectionVariable().getEntityId()));
+      computeInputVars.addAll(getChildrenVariables(computeConfig.getCollectionVariable()));
+      connection.voidEval(getVoidEvalFreadCommand(COMPUTE_STREAM_NAME,
+        computeInputVars
+      ));
+      connection.voidEval("alphaDivDT <- alphaDiv(" + COMPUTE_STREAM_NAME + ", " + 
+                                                      computeEntityPKColName + ", " + 
+                                                      valueSpec + ")");
+      connection.voidEval(getVoidEvalFreadCommand(DEFAULT_SINGLE_STREAM_NAME, 
+          spec.getXAxisVariable(),
+          spec.getOverlayVariable(),
+          getVariableSpecFromList(spec.getFacetVariable(), 0),
+          getVariableSpecFromList(spec.getFacetVariable(), 1)));
+      // merge alpha div and other viz stream data as input to boxplot
+      connection.voidEval("vizData <- merge(alphaDivDT, " + 
+                                            DEFAULT_SINGLE_STREAM_NAME + 
+                                         ", by=" + computeEntityPKColName +")");
+      connection.voidEval(getVoidEvalVarMetadataMap(DEFAULT_SINGLE_STREAM_NAME, varMap));
+      // update the new map obj in R to add alphaDiv
+      connection.voidEval("map <- rbind(map, list('id'='alphaDiversity'," +
+                                                 "'plotRef'='yAxisVariable'," +
+                                                 "'dataType'='NUMBER'," +
```

Comment:
data type and shape should probably come from computedVariableDetails. I'll add it!

Reply:
Updated in microbiomeComputations. `computedVariableDetails` now contains `variableId`, `entityId`, `dataType`, `dataShape` (among others).